### PR TITLE
Add "silent notifications" setting

### PR DESCRIPTION
**Pull request by asterite**
Issue: #327

### Requirements

- [ ]  issue was opened to discuss proposed changes before starting implementation.
- [ ]  during development, `node` version specified in `package.json` was used.
- [x]  package versions and package-lock.json were not changed (`npm install --no-save`).
- [x]  app version number was not changed.
- [ ]  all new code has tests to ensure against regressions.
- [ ] `npm run lint` reports no offenses.
- [x] `npm run test` is error-free.
- [x]  README and CHANGELOG were updated accordingly.

Explanations to why some things are unchecked:
- I opened an issue but forgot to discuss the implementation. I describe the implementation below. If you don't agree with it you can close this PR, or use it as a basis for another implementation.
- I couldn't find an easy way to install that specific node version (I actually have an older `node` version and then I upgraded to a newer `node` version, and in both cases it worked fine) but the changes are relatively small that it shouldn't break depending on the version
- I didn't add tests to the code, but it didn't have tests either
- `npm run lint` gives errors, but it gave errors before (I can fix this in a next PR if you want)

### Description of the Change

This PR adds a new setting to the 3rd settings page: "silent notifications". It's unchecked by default, so the default behavior is the same as before this PR. If checked, the OS notifications that pop-up are now silent.

I thought about doing this only when the "silent" theme is selected. However, making it in this way is more flexible because you can have a silent theme but sound-y notifications, or the other way around.

Additionally, ~in a separate commit~, I extracted the "show notification" logic to a separate function because some duplicated code started to show.

I added default English texts for the language translations expect for Spanish because I'm a native Spanish speaker (from Argentina). In a next PR I can send the missing Spanish translations if you want.

Oh, I unchecked one of the TODO checkboxes, but I'm not sure that's right. I can revert that change if needed.

### Verification Process

To verify that it worked I:
- left the new checkbox unchecked and confirmed that a sound appears when showing any of the notifications (break, microbreak and resume)
- left the new checkbox checked and confirmed that a sound **doesn't** appear when showing any of the notifications (break, microbreak and resume)

I assume the changes didn't introduce any regressions because they only modify that part of the app's logic. I also made sure to update all code that's related to the `showNotification` logic.

### Other information

Thank you for the app, it's great! The least I can do is try to improve it :-)

### CODE REVIEW

**Comment by asterite** [comment]
I will rebase in a few hours

File: app/utils/defaultSettings.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ module.exports = {
   posLatitude: 0.0,
   posLongitude: 0.0,
   useMonochromeTrayIcon: true,
+  silentNotifications: true,
```
Comment (hovancik) [inline app/utils/defaultSettings.js]: by default should be false, so we do not change the existing behavior

**Comment by hovancik** [comment]
Many thanks for PR ;] I will check the code. 

For different versions of node, I am using nvm https://github.com/creationix/nvm. 

Feel free to make Spanish translations better ;]

I have question: 

What about new version notification? Do you think we should also use this settings for it or not?

File: app/utils/defaultSettings.js (hunk 1 above)
Comment (asterite) [reply app/utils/defaultSettings.js]: Fixed!

**Comment by asterite** [comment]
Hi! I just rebased the code against `master`.

About the new version notification, I'd use the same preference. It's either all notifications on for the app or all off. That is, unless you want to have specific settings for each notification, but I think a single one is simpler. In general I think you either like the notification sound or you don't, globally for the app.

I'll make that change right away so it's consistent with everything else, but let me know if you'd like it some other way.

**Comment by hovancik** [comment]
I would do the same, but didn't want to influence you ;]

**Comment by asterite** [comment]
Done! Let me know what you think.

And thank you for the recommendation of `nvm`, it worked out of the box. Maybe you can suggest to use it in the issue template :-)

**Pull request merged**